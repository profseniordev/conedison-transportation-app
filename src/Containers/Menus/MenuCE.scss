@import '../../variables.scss';
.App-header {
  text-align: center;
  background-color: #ffffff;
  height: 72px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border-bottom: 1px solid #dbdede;
  font-size: 16px;
  font-family: Roboto Medium;
  .user-avatar {
    width: 40px;
    height: 40px;
    margin: 40px 16px;

    img {
      width: 100%;
    }
  }

  .user-info {
    line-height: 1.3;

    .fullname {
      font-size: 14px;
    }

    .role {
      color: #6f7780;
      font-size: 12px;
      font-family: Roboto Regular;
    }
  }

  .logo {
    font-size: 20px;
    font-family: 'Avenir Next Cyr Bold';
    color: #3a3c3e;
    margin-left: 35px;
  }

  .menu-web {
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 72px;
    .logo {
      width: 129px;
      height: 32px;
      margin: 0 30px 0 15px;
      background: url('../../Assets/logo.svg') no-repeat;
    }

    a {
      color: #1a242d;
      text-align: center;
      cursor: pointer;
      white-space: nowrap;
    }
    .menu-router {
      height: 72px;
      display: flex;
      align-items: center;
      justify-content: space-between;
      a {
        font-family: Roboto;
        font-style: normal;
        font-weight: normal;
        font-size: 16px;
        line-height: 24px;
        color: #1a242d;
        margin: 0;
        height: 100%;
        display: flex;
        align-items: center;
        transition: all 0.3s;
        &:before {
          content: '';
          height: 3px;
          width: 100%;
          border-radius: 21px 21px 0 0;
          position: absolute;
          bottom: 0;
          left: 0;
          transition: all 0.3s;
        }
        &.active {
          font-family: Roboto;
          font-style: normal;
          font-weight: 900;
          font-size: 16px;
          line-height: 24px;
          color: #0099d8 !important;
          background-color: white;
          text-decoration: none;
          position: relative;
          &:before {
            background: #0099d8;
          }
        }
      }
      a:hover:not(.active) {
        text-decoration: none;
        position: relative;
        transition: all 0.3s;
        &:before {
          background: #6f7780;
        }
      }
    }
    .header-action {
      a {
        position: relative;
        color: #6f7780a3;
        text-align: center;
      }

      svg path {
        fill: $grey !important;
      }

      .p-3 {
        height: 70px;
        display: flex;
        align-items: center;
      }

      a:hover:not(.active),
      .user-select:hover:not(.active) {
        background-color: #ddd;
        text-decoration: none;
        color: #666;
      }

      a.active {
        color: white !important;
        text-decoration: none;

        svg path {
          fill: #fff !important;
        }
      }
      .notifications-box {
        position: relative;
        .notifications {
          width: 48px;
          height: 48px;
          max-height: 48px;
          background: #ffffff;
          border: 1px solid #f2f2f2;
          box-shadow: 0px 6px 58px rgba(196, 203, 214, 0.103611);
          border-radius: 16px;
          margin-right: 32px;
        }
        .notification-toggle {
          position: absolute;
          z-index: 1000;
          top: 57px;
          right: 35px;
          width: 500px;

          &:before {
            content: '';
            position: absolute;
            top: 0;
            right: 0;
            transform: translate(-13px, -6px) rotate(45deg);
            border-top: 1px solid rgba(219, 222, 222, 0.41);
            border-left: 1px solid rgba(219, 222, 222, 0.41);
            width: 15px;
            height: 15px;
            background: #edf1f5;
          }
        }
      }

      .user-select-box {
        padding: 11px 0;
        position: relative;
        .user-select {
          height: 48px;
          width: auto;
          min-width: 180px;
          background: #ffffff;
          border: 1px solid #f2f2f2;
          box-shadow: 0px 6px 58px rgba(196, 203, 214, 0.103611);
          border-radius: 16px;
          position: relative;
          z-index: 1001;
          display: flex;
          align-items: center;
          cursor: pointer;
          .initials {
            width: 30px;
            height: 30px;
            background: rgba(47, 128, 237, 0.2);
            border: 2px solid #ffffff;
            border-radius: 50%;
            margin: 9px 12px;
            font-family: Roboto;
            font-style: normal;
            font-weight: bold;
            font-size: 12px;
            line-height: 24px;
            display: flex;
            align-items: center;
            justify-content: center;
            color: #2f80ed;
          }
          &:before {
            content: url(../../Assets/icons/caret_down.svg);
            height: 24px;
            width: 24px;
            position: absolute;
            right: 10px;
            top: 10px;
          }
        }
        .user-select-dropdown {
          width: 100%;
          margin-top: 11px;
          position: absolute;
          z-index: 1000;
          transform: translateY(-1000%);
          transition: all 0.3s ease-in-out;

          background: #fff;
          box-shadow: 2px 2px 6px rgba(#000, 0.1);
          &:before {
            content: '';
            position: absolute;
            top: 0;
            right: 4px;
            transform: translate(-13px, -7px) rotate(45deg);
            border-top: 1px solid rgba(219, 222, 222, 0.41);
            border-left: 1px solid rgba(219, 222, 222, 0.41);
            width: 15px;
            height: 15px;
            background: #fff;
          }
        }
        .profile-menu {
          position: absolute;
          z-index: 1000;
          top: 80px;
          right: 0.5rem;
          width: 350px;

          &:before {
            content: '';
            position: absolute;
            top: 0;
            right: 4px;
            transform: translate(-13px, -7px) rotate(45deg);
            border-top: 1px solid rgba(219, 222, 222, 0.41);
            border-left: 1px solid rgba(219, 222, 222, 0.41);
            width: 15px;
            height: 15px;
            background: #fff;
          }

          .box-item {
            background: #fff;
            box-shadow: 2px 2px 6px rgba(#000, 0.1);
          }

          a {
            transition: all 0.4s;
            background: transparent !important;
            color: $grey;
          }

          &__user {
            display: flex;
            justify-content: space-between;
            align-items: center;
            padding: 20px 24px;
            border-bottom: 1px solid #dbdede;
          }

          &__info {
            display: flex;

            .avatar {
              max-width: 40px;
            }

            img {
              max-width: 100%;
            }
          }

          &__name {
            padding-left: 6px;

            p {
              margin-bottom: 0;
            }

            .title {
              color: #3a3c3e;
              font-size: 14px;
              text-align: left;
            }

            .categories {
              color: $grey;
              font-size: 12px;
              text-align: left;
              display: flex;
              flex-direction: column;
            }
          }

          &__button {
            cursor: pointer;
            padding: 12px 24px;
            color: $grey;
            font-size: 14px;
            text-align: left;
            transition: all 0.3s;

            &:hover {
              color: #aaa;
            }
          }
        }
      }
      .user-select-box:hover {
        .user-select-dropdown {
          transform: translateY(0%);
        }
      }
    }
  }

  .menu-mobile {
    display: none;

    .menu-icon {
      height: 55px;
      width: 55px;
      display: none;
    }

    &__no-login {
      display: flex;
      align-items: center;
      justify-content: center;
      height: 69px;

      .logo {
        margin-left: 0;
      }
    }

    .icon-bell {
      padding: 13px 12px 16px !important;

      svg path {
        fill: $grey !important;
      }

      &.active {
        background: #009ada;

        svg path {
          fill: #fff !important;
        }
      }
    }
  }

  .ce-sidebar {
    display: none;
    height: 100%;
    width: 290px;
    background-color: #fff;
    position: fixed !important;
    z-index: 999999 !important;
    overflow: auto;
    transition: 1s;

    ul {
      list-style-type: none;
      margin: 0;
      padding: 0;
      overflow: hidden;
      border: none;
      background-color: #ffffff;
      width: 100%;

      li a {
        display: block;
        color: $grey;
        text-align: left;
        padding: 14px 16px;
        text-decoration: none;
      }
    }
  }

  .ce-sidebar-opacity {
    display: none;
    height: 100%;
    width: 100%;
    background-color: #616364;
    position: fixed !important;
    z-index: 111111;
    overflow: auto;
    transition: 1s;
    opacity: 0.9;
  }

  .log-out {
    position: absolute;
    right: 0;
    bottom: 0;
    left: 0;
    padding: 0 16px 50px 16px;
    text-align: center;
    float: none;

    a {
      display: flex !important;
      align-items: center;
      padding: 0 !important;
    }
  }

  .styleHr {
    margin-top: 0 !important;
    margin-bottom: 0 !important;
  }
}
.display-none{
  display: none;
}
@media screen and (max-width: 1123px) {
  .display{
    display: block;
  }
  .last-name-display-none{
    display: none;
  }
  .App-header .menu-web .menu-router a {
    font-size: 15px;
  }
  .App-header .menu-web .menu-router a.active{
    font-size: 15px;
  }
}
