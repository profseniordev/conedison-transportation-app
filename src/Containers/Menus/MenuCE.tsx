/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from 'react';
import { NavLink } from 'react-router-dom';
import './MenuCE.scss';
// import Cogwheel from '../../Images/cogwheel.png';
import Logout from '../../Images/logout.png';
import Right from '../../Images/chevron-right-12.png';
import NoAvatar from '../../Images/no-ava.png';
// import * as CeIcon from '../../Utils/Icon';
// import ProfileMenu from './../Notifications/ProfileMenu';
import { connect } from 'react-redux';
import { actions } from '../../Services';
import some from 'lodash/some';
import history from '../../history';
import ProfileMenus from './ProfileMenus/ProfileMenus';

interface Props {
  logout: () => void;
  notifications: any;
  user: any;
  has_access_to_dispatch: boolean;
  canAccessWorker: boolean;
  canAccessSubContractor: boolean;
  canAccessInvoice: boolean;
  canAccessRoles: boolean;
  canAccessReports: boolean;
}

const MenuCE: React.FC<Props> = ({
  logout,
  notifications,
  user,
  has_access_to_dispatch,
  canAccessWorker,
  canAccessSubContractor,
  canAccessInvoice,
  canAccessRoles,
  canAccessReports,
}) => {
  const [haveUnreadNotifications, setHaveUnreadNotifications] = useState(false);

  // const [toggleProfile, setToggleProfile] = useState(false);

  // const toggleProfileMenu = () => {
  //   setToggleProfile(!toggleProfile);
  // };

  const ClickToMenu = () => {
    const el: HTMLElement | null = document.getElementById('ceSidebar');
    if (el) {
      const definitelyAnElement: HTMLElement = el;
      definitelyAnElement.style.display = 'block';
    }
    const el1: HTMLElement | null = document.getElementById('ceSidebarOpacity');
    if (el1) {
      const definitelyAnElement1: HTMLElement = el1;
      definitelyAnElement1.style.display = 'block';
    }
  };

  const ClickToCloseMenu = () => {
    const el: HTMLElement | null = document.getElementById('ceSidebar');
    if (el) {
      const definitelyAnElement: HTMLElement = el;
      definitelyAnElement.style.display = 'none';
    }
    const el1: HTMLElement | null = document.getElementById('ceSidebarOpacity');
    if (el1) {
      const definitelyAnElement1: HTMLElement = el1;
      definitelyAnElement1.style.display = 'none';
    }
  };

  const onLogout = () => {
    logout();
    ClickToCloseMenu();
  };

  const navigate = (link) => {
    const urlParams = new URLSearchParams(window.location.search);
    const jobId = urlParams.get('jobId');

    if (link === '/timesheets' && jobId) {
      window.location.replace(link);
    } else {
      history.push(link);
    }
  };

  const checkUnreadNotification = () => {
    setHaveUnreadNotifications(
      some(notifications, {
        isRead: false,
      })
    );
  };

  useEffect(() => {
    checkUnreadNotification();
  }, [notifications]);

  const header = () => (
    <header className="App-header">
      <div>
        <div id="ceSidebar" className="ce-sidebar">
          <div className="d-flex justify-content-between align-items-center">
            <div className="d-flex justify-content-between align-items-center">
              <div className="user-avatar">
                <img src={NoAvatar} alt="" />
              </div>
              <div className="user-info text-left">
                <div className="fullname">
                  {user && `${user.firstName} ${user.lastName}`}
                </div>
                <div className="role">
                  {user && user.roles && user.roles.join(', ')}
                </div>
              </div>
            </div>
            <div className="mr-3">
              <img src={Right} alt="" />
            </div>
          </div>
          <hr className="styleHr" />
          <ul className="mt-3">
            {has_access_to_dispatch && (
              <li>
                <span
                  className="link cursor-pointer"
                  onClick={(_) => navigate('/dispatch')}
                >
                  Dispatch
                </span>
              </li>
            )}
            <li>
              <span
                className="link cursor-pointer"
                onClick={(_) => navigate('/map')}
              >
                Map
              </span>
            </li>
            <li>
              <span
                className="link cursor-pointer"
                onClick={(_) => navigate('/job')}
              >
                Job List
              </span>
            </li>
            <li>
              <span
                className="link cursor-pointer"
                onClick={(_) => navigate('/job-grid')}
              >
                Job Grid
              </span>
            </li>
            {canAccessWorker && (
              <li>
                <span
                  className="link cursor-pointer"
                  onClick={(_) => navigate('/workers')}
                >
                  Workers
                </span>
              </li>
            )}
            {canAccessSubContractor && (
              <li>
                <span
                  className="link cursor-pointer"
                  onClick={(_) => navigate('/subcontractors')}
                >
                  Subcontractors
                </span>
              </li>
            )}
            <li>
              <span
                className="link cursor-pointer"
                onClick={(_) => navigate('/timesheets')}
              >
                Timesheets
              </span>
            </li>
            {canAccessInvoice && (
              <li>
                <span
                  className="link cursor-pointer"
                  onClick={(_) => navigate('/invoices')}
                >
                  Invoices
                </span>
              </li>
            )}
            {has_access_to_dispatch && (
              <li>
                <span
                  className="link cursor-pointer"
                  onClick={(_) => navigate('/billing')}
                 >
                  Billing
                </span>
              </li>
            )}
            {canAccessReports && (
              <li>
                <span
                  className="link cursor-pointer"
                  onClick={(_) => navigate('/reports')}
                >
                  Reports
                </span>
              </li>
            )}
            {canAccessRoles && (
              <li>
                <span
                  className="link cursor-pointer"
                  onClick={(_) => navigate('/roles')}
                >
                  Settings
                </span>
              </li>
            )}
            <li className="log-out" onClick={onLogout}>
              <span>
                <img className="mr-2" src={Logout} alt="" />
                <span>Log Out</span>
              </span>
            </li>
          </ul>
        </div>
        <div
          onClick={() => {
            ClickToCloseMenu();
          }}
          id="ceSidebarOpacity"
          className="ce-sidebar-opacity"
        />
        <div className="menu-mobile">
          <div
            className="menu-icon d-flex align-items-center"
            onClick={ClickToMenu}
          >
            <img
              className="m-auto"
              alt=""
              src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAA4SURBVFhH7dOxCQAADAJB9186WUCwMZDiD760VAA+muMiN2oGIHLXaRa5UTMAkbtOs8iNmgH4RFrzpY9xuWJl5QAAAABJRU5ErkJggg=="
            />
          </div>
          <img
            src={
              'https://storage.googleapis.com/conedison/assets/coned_logo.png'
            }
            alt="logo"
            style={{ width: 143, height: 24 }}
          />

          {/* {toggleProfile && <ProfileMenu />} */}
        </div>

        <div className="menu-web">
          <img
            src={
              'https://storage.googleapis.com/conedison/assets/coned_logo.png'
            }
            alt="logo"
            className="d-none d-md-block ml-3 "
            style={{ width: 143, height: 24 }}
          />
          <div className="menu-router">
            {has_access_to_dispatch && (
              <NavLink className=" p-lg-2 p-md-1 p-xl-3" id="dispatch" to="/dispatch">
                Dispatch
              </NavLink>
            )}
            <NavLink className="p-lg-2 p-md-1 p-xl-3" id="map" to="/map">
              Map
            </NavLink>
            <NavLink className="p-lg-2 p-md-1 p-xl-3" id="job" to="/job">
              Job List
            </NavLink>
            <NavLink className="p-lg-2 p-md-1 p-xl-3" id="job-grid" to="/job-grid">
              Job Grid
            </NavLink>
            {canAccessWorker && (
              <NavLink className="p-lg-2 p-md-1 p-xl-3" id="workers" to="/workers">
                Workers
              </NavLink>
            )}
            {canAccessSubContractor && (
              <NavLink
                className="p-lg-2 p-md-1 p-xl-3"
                id="subcontractors"
                to="/subcontractors"
              >
                <span>Subcontractors</span>
              </NavLink>
            )}
            <NavLink
              className="p-lg-2 p-md-1 p-xl-3"
              id="timesheets"
              to="/timesheets"
              onClick={() => navigate('/timesheets')}
            >
              Timesheets
            </NavLink>
            {has_access_to_dispatch && (
              <NavLink className="p-lg-2 p-md-1 p-xl-3" id="billing" to="/billing">
                Billing
              </NavLink>
            )}
            {canAccessInvoice && (
              <NavLink className="p-lg-2 p-md-1 p-xl-3" id="invoices" to="/invoices">
                Invoices
              </NavLink>
            )}
            {canAccessReports && (
              <NavLink
                className="p-lg-2 p-md-1 p-xl-3"
                id="reports"
                to="/reports"
                onClick={() => navigate('/reports')}
              >
                Reports
              </NavLink>
            )}
          </div>
          <div className="header-action d-flex align-items-center mr-3">
            {/* <NavLink
              style={{ borderBottom: 'none' }}
              className={`p-3 notification-icon ${
                haveUnreadNotifications ? 'active' : ''
              }`}
              to="/notifications"
            >
              <div className="border-custom">
                <CeIcon.BellIcon />
              </div>
            </NavLink> */}
            {/* {canAccessRoles && (
              <NavLink
                style={{ borderBottom: 'none' }}
                className="p-3"
                id="roles"
                to="/roles"
              >
                <div className="border-custom">
                  <img src={Cogwheel} alt="" />
                </div>
              </NavLink>
            )} */}
            {/* <div
              className={`p-3 user-icon ${toggleProfile ? 'active' : ''}`}
              id="profile"
              onClick={() => toggleProfileMenu()}
            >
              <div className="border-custom">
                <CeIcon.UserIcon />
              </div>
            </div> */}

            {/* {toggleProfile && <ProfileMenu />} */}

            <ProfileMenus />
          </div>
        </div>
      </div>
    </header>
  );

  return user == null ? null : header();
};

function mapStateToProps({ app, notifications }) {
  return {
    canAccessWorker: app.user
      ? [5, 6, 8].some((r) => app.user.roles.includes(r))
      : false,
    canAccessSubContractor: app.user
      ? [5, 6, 8].some((r) => app.user.roles.includes(r))
      : false,
    canAccessInvoice: app.user
      ? [4, 7, 8].some((r) => app.user.roles.includes(r))
      : false,
    canAccessRoles: app.user
      ? [6, 8].some((r) => app.user.roles.includes(r))
      : false,
    canAccessReports: app.user
      ? [1, 2, 3, 4, 7, 8].some((r) => app.user.roles.includes(r))
      : false,
    has_access_to_dispatch: app.user
      ? ![1, 2, 3, 4].some((r) => app.user.roles.includes(r))
      : false,
    user: app.user,
    notifications: notifications.notifications,
  };
}
function mapDispatchToProps(dispatch) {
  return {
    dispatch,
    logout: () => dispatch(actions.AppActions.logout()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(MenuCE);
