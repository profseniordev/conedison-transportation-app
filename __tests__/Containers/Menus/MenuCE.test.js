import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import MenuCE from '../../../src/Containers/Menus/MenuCE';

const mockStore = configureMockStore();
const store = mockStore({
  app: {
    user: {
      roles: [1],
    },
  },
  notifications: {
    notifications: [],
  },
});

const MockLink = ({ id, className, to, children }) => {
  return (
    <a className={className} id={id} href={to}>
      {children}
    </a>
  );
};

jest.mock('react-router-dom', () => ({
  __esModule: true,
  useLocation: jest.fn().mockReturnValue({
    pathname: '/job',
    search: '',
    hash: '',
    state: null,
    key: '5nvxpbdafa',
  }),
  NavLink: (props) => <MockLink {...props} />,
}));

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MenuCE />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
