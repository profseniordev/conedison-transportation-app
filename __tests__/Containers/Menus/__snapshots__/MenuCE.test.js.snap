// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="App-header"
>
  <div>
    <div
      className="ce-sidebar"
      id="ceSidebar"
    >
      <div
        className="d-flex justify-content-between align-items-center"
      >
        <div
          className="d-flex justify-content-between align-items-center"
        >
          <div
            className="user-avatar"
          >
            <img
              alt=""
              src="no-ava.png"
            />
          </div>
          <div
            className="user-info text-left"
          >
            <div
              className="fullname"
            >
              undefined undefined
            </div>
            <div
              className="role"
            >
              1
            </div>
          </div>
        </div>
        <div
          className="mr-3"
        >
          <img
            alt=""
            src="chevron-right-12.png"
          />
        </div>
      </div>
      <hr
        className="styleHr"
      />
      <ul
        className="mt-3"
      >
        <li>
          <span
            className="link cursor-pointer"
            onClick={[Function]}
          >
            Map
          </span>
        </li>
        <li>
          <span
            className="link cursor-pointer"
            onClick={[Function]}
          >
            Job List
          </span>
        </li>
        <li>
          <span
            className="link cursor-pointer"
            onClick={[Function]}
          >
            Job Grid
          </span>
        </li>
        <li>
          <span
            className="link cursor-pointer"
            onClick={[Function]}
          >
            Timesheets
          </span>
        </li>
        <li
          className="log-out"
          onClick={[Function]}
        >
          <span>
            <img
              alt=""
              className="mr-2"
              src="logout.png"
            />
            <span>
              Log Out
            </span>
          </span>
        </li>
      </ul>
    </div>
    <div
      className="ce-sidebar-opacity"
      id="ceSidebarOpacity"
      onClick={[Function]}
    />
    <div
      className="menu-mobile"
    >
      <div
        className="menu-icon d-flex align-items-center"
        onClick={[Function]}
      >
        <img
          alt=""
          className="m-auto"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAA4SURBVFhH7dOxCQAADAJB9186WUCwMZDiD760VAA+muMiN2oGIHLXaRa5UTMAkbtOs8iNmgH4RFrzpY9xuWJl5QAAAABJRU5ErkJggg=="
        />
      </div>
      <img
        alt="logo"
        src="https://storage.googleapis.com/conedison/assets/coned_logo.png"
        style={
          Object {
            "height": 24,
            "width": 143,
          }
        }
      />
    </div>
    <div
      className="menu-web"
    >
      <img
        alt="logo"
        className="d-none d-md-block ml-3 "
        src="https://storage.googleapis.com/conedison/assets/coned_logo.png"
        style={
          Object {
            "height": 24,
            "width": 143,
          }
        }
      />
      <div
        className="menu-router"
      >
        <a
          className="p-xl-3"
          href="/map"
          id="map"
        >
          Map
        </a>
        <a
          className="p-xl-3"
          href="/job"
          id="job"
        >
          Job List
        </a>
        <a
          className="p-xl-3"
          href="/job-grid"
          id="job-grid"
        >
          Job Grid
        </a>
        <a
          className="p-xl-3"
          href="/timesheets"
          id="timesheets"
        >
          Timesheets
        </a>
      </div>
      <div
        className="header-action d-flex align-items-center mr-3"
      >
        <a
          className="p-3 notification-icon "
          href="/notifications"
        >
          <div
            className="border-custom"
          >
            <svg>
              bell.svg
            </svg>
          </div>
        </a>
        <div
          className="p-3 user-icon "
          id="profile"
          onClick={[Function]}
        >
          <div
            className="border-custom"
          >
            <svg>
              user.svg
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
